api/utils/marshamallow_schemas.py

```diff
@@ -0,0 +1,103 @@
+"""Validation schemas."""
+from marshmallow import (Schema, fields, post_load, validate,
+                         ValidationError)
+from api.models import User, Activity, ActivityType
+
+
+class BaseSchema(Schema):
+    """Creates a base validation schema."""
+
+    uuid = fields.String(dump_only=True, validate=[
+        validate.Length(max=36)])
+    name = fields.String(
+        required=True,
+        error_messages={
+            'required': {'message': 'A name is required.',
+                         'status_code': 400}
+        })
+    photo = fields.String()
+    created_at = fields.DateTime(dump_only=True)
+    modified_at = fields.DateTime(dump_only=True)
+    description = fields.String()
+
+    @post_load
+    def verify_activity_type(self, data):
+        """Verify if activity type exists."""
+        activity_type = ActivityType.query.filter(
+            ActivityType.uuid.ilike(data['activity_type_id'])).first()
+        existing_activity = Activity.query.filter(
+            Activity.name.ilike(data['name'])).first()
+
+        if not activity_type:
+            raise ValidationError(
+                {'message': 'Activity Type does not exist!',
+                 'status_code': 404})
+
+        if existing_activity:
+            raise ValidationError(
+                {'message': 'Activity already exists!',
+                 'status_code': 405})
+
+
+class ActivityTypesSchema(BaseSchema):
+    """Creates a validation schema for activity types."""
+
+    description = fields.String(
+        required=True,
+        error_messages={
+            'required': 'A description is required.'
+        }
+    )
+    value = fields.Integer(
+        required=True,
+        error_messages={
+            'required': 'Please send the activity points value'
+        }
+    )
+
+
+class LoggedActivitySchema(BaseSchema):
+    """Creates a validation schema for logged activities."""
+
+    status = fields.String(dump_only=True)
+    value = fields.Integer(
+        required=True,
+        error_messages={
+            'required': 'Please send the activity points value'
+        }
+    )
+    user = fields.String(dump_only=True, attribute='user.name')
+    society = fields.String(dump_only=True, attribute='society.name')
+    approved_by = fields.Method('get_approver', dump_only=True)
+
+    @staticmethod
+    def get_approver(obj):
+        """Get approver details."""
+        if obj.approver_id:
+            approver = User.query.get(obj.approver_id)
+            return approver.name
+        else:
```

Comment:
Could you also remove the else for men here? I noticed it's a bit superfluous. My bad. You can just leave in the return alone.  Like this:
```python
    return approver.name
return
```

Reply:
Done.